### ISSUE TITLE
DefaultProps not replacing undefined function props

### ISSUE THREAD

**Issue by yormi** [facebook/react#7562]
**BUG**

**What is the current behavior?**
The default function A given in `static defaultProps = { A: () => {...} }` is not replacing a function prop that is not defined while the defaultProps works for a string.

**Minimal example**
https://jsfiddle.net/pLp2fbvq/

**What is the expected behavior?**
When calling `this.props.A()` no `Uncaught TypeError: this.props.A is not a function` appears and it returns the value returned by the defaultProp `A`.

**Versions**
I tried with 15.1.0 and 15.3.1 and had the same result. Don't know about the other versions.

**Comment by bloodyowl** [comment]
Weird, can you file a jsfiddle with a minimal example of this behaviour?

**Comment by yormi** [comment]
... I forgot to save it apparently.
I tried to do it again and understood my mistake.
Sorry for the trouble.

For those making the same mistake:
I defined my function after the class definition. Since the static defaultProps is evaluated before it was undefined.

Moreover, when I made the minimal example in jsfiddle I think I end up using this: (note the colon instead of the equal sign)
```javascript
static defaultProps: {...}
```
The defaultProps were just ignored. When I wanted to redo it, it noticed that the following throws because of the "=" in jsfiddle... it should right ?
```javascript
static default = {...}
```
